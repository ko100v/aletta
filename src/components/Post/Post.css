.title {
  font-size: 3rem;
}

.content {
  font-size: 1.8rem;
  line-height: 2;
  word-wrap: break-word;
}

.link {
  color: #2F89DE;
}

.link:focus,
.link:hover {
  color: #30699e ;
  transition: color 0.3s ease;
}