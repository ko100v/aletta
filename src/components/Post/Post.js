import React, { Component } from 'react'
import styles from './Post.css'

import { Link } from 'react-router-dom'

class Post extends Component {

  render() {
    const { title, content, link } = styles

    return (
      <div>
        <h2 className={title}>
          {this.props.titleLink
            ? <Link className={styles.link} to={`/post/${this.props.slug}`}>{this.props.title}</Link>
            : this.props.title}
        </h2>
        <p className={content}>{this.props.excerpt}</p>
      </div>
    )
  }
}

export default Post