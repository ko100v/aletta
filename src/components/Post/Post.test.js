import React from 'react'

import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

import Post from './Post'

test('Render Post <p />', () => {
    const component = shallow(<Post title='My Post' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
})

test('Render Post <Link /> ', () => {
    const component = shallow(<Post title='My Post' titleLink />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
})