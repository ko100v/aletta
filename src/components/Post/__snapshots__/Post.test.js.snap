// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Post <Link />  1`] = `
<div>
  <h2
    className="title"
  >
    <Link
      className="link"
      replace={false}
      to="/post/undefined"
    >
      My Post
       
    </Link>
  </h2>
  <p
    className="content"
  />
</div>
`;

exports[`Render Post <p /> 1`] = `
<div>
  <h2
    className="title"
  >
    My Post
  </h2>
  <p
    className="content"
  />
</div>
`;
